### PR TITLE
chirpstack-concentratord: init at 4.4.2

### PR DESCRIPTION
**Pull request by stv0g**
Package Chirpstack Concentrator HAL daemon for LoRa gateways and related libraries.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ch/chirpstack-concentratord/package.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  ];
+
+  passthru.updateScript = gitUpdater { rev-prefix = "v"; };
+
+  meta = {
+    description = "Concentrator HAL daemon for LoRa gateways";
+    homepage = "https://www.chirpstack.io/";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.stv0g ];
+    platforms = [ "aarch64-linux" ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ch/chirpstack-concentratord/package.nix:42]: Why is this only available on aarch64-linux?

Comment (Aleksanaa) [reply pkgs/by-name/ch/chirpstack-concentratord/package.nix:42]: Specific hardware?

File: pkgs/by-name/li/libloragw-2g4/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  fetchFromGitHub,
+}:
+stdenv.mkDerivation {
+  pname = "libloragw-2g4";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "Lora-net";
+    repo = "gateway_2g4_hal";
+    rev = "f14f5cf2e4caf3789bc32159fba5c10363166591";
```
Comment (Aleksanaa) [inline pkgs/by-name/li/libloragw-2g4/package.nix:13]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/li/libloragw-sx1301/package.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  fetchFromGitHub,
+}:
+stdenv.mkDerivation {
+  pname = "libloragw-sx1301";
+  version = "5.0.1r2";
+
+  src = fetchFromGitHub {
+    owner = "brocaar";
+    repo = "lora_gateway";
+    rev = "59381129a07858a2a91aeffe21cd6a728219cf23";
```
Comment (Aleksanaa) [inline pkgs/by-name/li/libloragw-sx1301/package.nix:13]: Also

File: pkgs/by-name/li/libloragw-sx1302/package.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  fetchFromGitHub,
+}:
+stdenv.mkDerivation {
+  pname = "libloragw-sx1302";
+  version = "2.1.0r7";
+
+  src = fetchFromGitHub {
+    owner = "brocaar";
+    repo = "sx1302_hal";
+    rev = "c3d99009556fdfe273c3a53306082ef181333c7a";
```
Comment (Aleksanaa) [inline pkgs/by-name/li/libloragw-sx1302/package.nix:13]: Also

File: pkgs/by-name/li/libloragw-sx1301/package.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    hash = "sha256-K+GCE3eJq/9CgjIPTCBx3dSbDP0QbTiv0QMAbbLZM7s=";
+  };
+
+  makeFlags = [
+    "-e"
+    "-C"
+    "libloragw"
+  ];
+
+  installPhase = ''
```
Comment (Aleksanaa) [inline pkgs/by-name/li/libloragw-sx1301/package.nix:23]: runHook

File: pkgs/by-name/li/libloragw-sx1301/package.nix (hunk 3 above)
Comment (stv0g) [reply pkgs/by-name/li/libloragw-sx1301/package.nix]: Fixed in e757e6f3c2ef

File: pkgs/by-name/li/libloragw-sx1301/package.nix (hunk 5 above)
Comment (stv0g) [reply pkgs/by-name/li/libloragw-sx1301/package.nix:26]: Fixed in 1ec52f0b996b

File: pkgs/by-name/li/libloragw-sx1302/package.nix (hunk 4 above)
Comment (stv0g) [reply pkgs/by-name/li/libloragw-sx1302/package.nix]: Fixed in 1ec52f0b996b

File: pkgs/by-name/ch/chirpstack-concentratord/package.nix (hunk 1 above)
Comment (stv0g) [reply pkgs/by-name/ch/chirpstack-concentratord/package.nix]: Its a LoRaWAN gateway which I have been testing on a ARM64 single-board computer.
But I checked, that in fact its just using Linux's SPI/I2C drivers.
So it also works on other architectures.

I tested the build now on:
- aarch64-linux
- x86_64-linux

Fixed in 1ec52f0b996b

**Pull request merged**